### PR TITLE
implements bifoldable test

### PR DESCRIPTION
**Pull request by solomon-b**
Resolves issue #48. We could reduce the number of type parameters by merging `c` and `m`, but I didn't want it to appear that `bifoldr f g z t ≡ appEndo (bifoldMap (Endo . f) (Endo . g) t) z` requires a monoid on `c`

### CODE REVIEW

File: src/Test/QuickCheck/Classes.hs [hunk 1]
```diff
@@ -826,3 +827,26 @@ foldableFunctor = const ( "Foldable Functor"
...
     foldMapP f t = foldMap f t =-= fold (fmap f t)
+
+bifoldable :: forall p a b c m.
+               ( Bifoldable p, Monoid m
+               , Show (p a b), Show (p m m)
+               , Arbitrary (p a b), Arbitrary (p m m), Arbitrary m
+               , CoArbitrary a, CoArbitrary b
+               , EqProp m, EqProp c, CoArbitrary c, Arbitrary c, Show c) =>
+               p a (b, c, m)  -> TestBatch
+bifoldable = const ( "traversable"
```
Comment (sjakobi) [inline src/Test/QuickCheck/Classes.hs:838]:
```suggestion
bifoldable = const ( "Bifoldable"
```

**Review by sjakobi — COMMENTED** [review]
Thanks! :)

Do you have a source for these laws?

**Comment by solomon-b** [comment]
They come from the Data.Bifoldable [haddocks](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Bifoldable.html#t:Bifoldable).

There is one more law I forget to include:

```
bifoldMap f g = bifold . bimap f g
```

This is only relevant if you have a Bifunctor. Would we put it in a separate test?

**Comment by sjakobi** [comment]
> There is one more law I forget to include:
> 
> ```
> bifoldMap f g = bifold . bimap f g
> ```
> 
> This is only relevant if you have a Bifunctor. Would we put it in a separate test?

Yes, we can add this in the style of the existing `foldableFunctor` test.

---

Regarding compatibility with GHC 7.10 and 8.0, I'll try to drop support for these soon, so we can merge your PRs without breaking CI.

**Comment by sjakobi** [comment]
Could you rebase?

**Comment by solomon-b** [comment]
> Could you rebase?

done!

**Pull request merged**